src/services/TemplatingService.ts

```diff
@@ -0,0 +1,150 @@
+  // region public static methods
+  // endregion
+
+  // region private static methods
+
+  private static insertMessagePayload(template: any, payload: any, platformId: string): any {
+    const config = Utils.getPlatformConfig(platformId);
+    return {
+      from: config.mail.defaults.from,
+      data: template.data,
+    };
+  }
+
+  private static insertLanguagePayload(template: any, payload: any): any {
+    // TODO: Insert message content into template
+    return Object.assign(template, {
+      subject: payload.title,
+      text: payload.body,
+      html: payload.body,
+      notification: {
+        title: payload.title,
+        body: payload.body,
+      },
+      android: template.android,
+      webpush: template.webpush,
+      apns: {
+        payload: {
+          aps: {
+            title: payload.title,
+            body: payload.body,
+          },
+        },
+      },
+    });
+  }
+
+  private static insertUserPayload(template: any, user: UserRessource): any {
+    // TODO: Access deviceId of User
+    return Object.assign(template, {
+      to: `${user.name} <${user.mail}>`,
+      token: 'user.deviceToken',
+    });
+  }
+
+  // endregion
+
+  // region public members
+  // endregion
+
+  // region private members
+
+  private readonly localizedTemplates: Template[] = [];
+
+  // endregion
+
+  // region constructor
+
+  public constructor(platformId: string, templateId: string, payload: any) {
+    this.generateTemplates(platformId, templateId, payload);
+  }
+
+  // endregion
+
+  // region public methods
+
+  public createMailMessage(user: UserRessource): Mail {
+    const template = this.getTemplateForUser(user, MAIL_MESSAGE);
+    const mail = {
+      from: template.from,
+      to: user.mail,
+      subject: template.subject,
+      text: template.text,
+      html: template.html,
+    };
+    return mail;
+  }
+
+  public createPushMessage(user: UserRessource): firebaseMessaging.Message {
+    const template = this.getTemplateForUser(user, PUSH_MESSAGE);
+    const push = {
+      token: template.token,
+      data: template.data,
+      notification: template.notification,
+      android: template.android,
+      webpush: template.webpush,
+      apns: template.apns,
+    };
+    return push;
+  }
+
+  // endregion
+
+  // region private methods
+
+  private generateTemplates(platformId: string, templateId: string, payload: TemplatePayload[]) {
+    for (const type of [MAIL_MESSAGE, PUSH_MESSAGE]) {
+      // Step 1: Load base template
+      let baseTemplate = Utils.getTemplate(platformId, templateId, type);
+
+      // Step 2: Insert general payload into base template
+      // TODO: Pass general message payload (with e.g. unique language identifier)
+      baseTemplate = TemplatingService.insertMessagePayload(baseTemplate, {}, platformId);
+
+      // Step 3: Insert language specific payload into base template for every language
+      for (const languagePayload of payload) {
```

Comment:
This could be extracted into an own method like `generateLanguagePaylaods` for better readablilty.

Reply:
Done